src/Silverback.Integration.Kafka/Messaging/Configuration/KafkaBrokerOptionsConfigurator.cs

```diff
@@ -35,7 +35,9 @@ public void Configure(IBrokerOptionsBuilder brokerOptionsBuilder)
                 .AddSingleton<IBrokerLogEnricher<KafkaProducerEndpoint>, KafkaLogEnricher>()
                 .AddSingleton<IBrokerLogEnricher<KafkaConsumerEndpoint>, KafkaLogEnricher>()
                 .AddSingleton<IBrokerActivityEnricher<KafkaProducerEndpoint>, KafkaActivityEnricher>()
-                .AddSingleton<IBrokerActivityEnricher<KafkaConsumerEndpoint>, KafkaActivityEnricher>();
+                .AddSingleton<IBrokerActivityEnricher<KafkaConsumerEndpoint>, KafkaActivityEnricher>()
+                .AddSingleton<IBrokerHeadersEnricher<KafkaProducerEndpoint>, KafkaBrokerHeadersEnricher>()
+                .AddSingleton<IBrokerHeadersEnricher<KafkaConsumerEndpoint>, KafkaBrokerHeadersEnricher>();
```

Comment:
Only the `KafkaProducerEndpoint` is needed.

Reply:
Uhm according to the e2e test it seems that only the KafkaConsumerEndpoint is needed.

Reply:
Do you have any idea regarding this issue?